_posts/2022-08-23-getting-around-7-day-cookie.markdown

```diff
@@ -0,0 +1,98 @@
+---
+layout: epic
+title: "Hacking Around Safari's 7-day Cookie Limit"
+subtitle: What we learned while trying to fix our cookie consent banner
+date: 2022-08-23
+categories: [Safari Cookies Consent Banner]
+author: [chris]
+---
+
+Amongst the many, many things that organizations have to contend with around
+cookie consent laws is Apple's very own browser, Safari. Did you know that
+Safari will only retain a client-side cookie for 7 days? This is in support of
+Apple's [Intelligent Tracking Prevention (ITP)][ITP] feature, designed to
+protect a user's privacy.
+
+These privacy efforts are great but, in hand with laws like [GDPR][] and
+[CCPA][], their rollout often creates a UX nightmare for users without some
+extra care. Here at Artsy, we've landed on a way to make things slightly less
+bad and want to share our approach.
+
+<!-- more -->
+
+Scenario: Imagine that as a EU resident you visit artsy.net for the first time.
+A banner appears asking you to `Accept` or `Deny` tracking cookies from our
+site. You don't like tracking cookies, so you click the "Deny" button and the
+banner disappears. All good, right? Nope! You visit Artsy a week later and
+again, a banner appears asking you to choose your preferences. This happens again
+and again until you switch browsers and realize that what you were experiencing
+was Apple's ITP feature in action. After choosing your preferences, the cookie we
+use to store them is erased after 7 days, necessitating another interaction.
+
+While pondering how to work around this UX issue, we landed on a simple solution
+prompted by a conversation with a Safari Engineer at Apple. She mentioned that
```

Comment:
Also note in my last comment that I suggested a change from Safari to WebKit as the kind of engineer we talked to. There's a bit of a difference.

Reply:
Oh, and that developers can schedule calls in open lab times. I think people don't know that and if we're sharing how to do things...